wikiclue/src/routes/levels/+page.svelte

```diff
@@ -0,0 +1,140 @@
+    import { currentLevel, levelWords } from '../../store/gameplay';
+    import { goto } from '$app/navigation';
+
+    const isOverlayOpen = writable(false);
+    const levelOver = writable(false);
+
+    let wordsToFind = ["",""];
+    let searchTerm = "";
+    let incorrectAnswer = false;
+
+    onMount(() => {
+        // When we use the database we need to get the user's current level instead
+		loadLevelWords();
+	});
+
+    function loadLevelWords() {
+        // Load the current level words from firebase, it's just in store/gameplay for now
+        wordsToFind[0] = levelWords[$currentLevel-1][0];
+        wordsToFind[1] = levelWords[$currentLevel-1][1];
+    }
+
+
+    function confirmPressed() {
+        incorrectAnswer = false;
+        // will need to change the if statement to use actual wikipedia api function
+        if (searchTerm.includes(wordsToFind[0]) && searchTerm.includes(wordsToFind[1])) {
+			endLevel();
+        } else {
+            incorrectAnswer = true;
+            setTimeout(() => {
+                incorrectAnswer = false;
+            }, 2000);
+        }
+    }
+
+    function storeData() {
+		// Save level data into database
+	}
+
+    function endLevel() {
+        storeData;
+        isOverlayOpen.set(true);
+		levelOver.set(true);
+	}
+
+    function playNextLevelClicked() {
+        // Reset when we get to the end for now
+        if ($currentLevel === 10) {
+            currentLevel.set(1);
+        } else {
+            currentLevel.update(n => n + 1);
+        }
+        loadLevelWords();
+        searchTerm= "";
+        levelOver.set(false);
+        isOverlayOpen.set(false);
+	}
+
+    function returnToMainMenuClicked() {
+        currentLevel.set(1);
+        goto('/home');
+	}
+
+    function onEnterPressed(event: KeyboardEvent) {
+        if (event.key === "Enter" && $levelOver) {
+            playNextLevelClicked()
+            return;
+        }
+        if (event.key === "Enter" && !$levelOver) {
+            confirmPressed();
+            return;
+        }
+    }
+</script>
+
+<svelte:window on:keydown={onEnterPressed} />
+
+<HeaderBar />
+<div class="levels-page">
+    <h1>Level Mode</h1>
+    <div class="info-bar">
+        <div class="info-container">
+            <LevelsIcon style="font-size: 2rem; color: black;"/>
+            <p class="info-text">
+                <span class="level-text">Current Level:</span>
+                <span class="streak-content">{$currentLevel}</span>
+            </p>
+        </div>
+    </div>
+    <div class="search-container">
+        <p class = "sub-header">Find a Wikipedia page that contains the following words:</p>
+        <div class="words-container">
+            <p class="search-words">{wordsToFind[0]}</p>
+            <p class="search-words">{wordsToFind[1]}</p>
+        </div>
+        {#if !$levelOver}
+            <input type="text" class="search-bar" placeholder="Enter the Wikipedia URL here..." bind:value={searchTerm}/>
```

Comment:
Let's change this since we're not doing urls anymore

Reply:
done